src/Chapter2.hs

```diff
@@ -841,7 +855,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate a l 
+  | a < 0 = []
+  | otherwise = take (length l) (drop a (cycle l))
```

Comment:
Great!
You also can implement the function more efficiently on some corner cases 🙂 For example, to check if the list is empty, and if `a` is equal to zero (by mod of the list's length) 🚀